### PR TITLE
Add deinterlace on VA module.

### PR DESCRIPTION
**Pull request by feiwan1**
Jira: None.
Test: Build passes on Linux and Android.

Signed-off-by: Wang,Fei <fei.w.wang@intel.com>

### CODE REVIEW

**Comment by feiwan1** [comment]
Rebase deinterlace code from https://github.com/intel/IA-Hardware-Composer/pull/199. Please have review. Thanks.

**Comment by kalyankondapally** [comment]
@feiwan1 SOrry need to rebase again :(

File: common/compositor/va/varenderer.cpp [hunk 1]
```diff
@@ -105,6 +105,12 @@ bool VARenderer::SetVAProcFilterColorDefaultValue(
   return true;
 }
 
+bool VARenderer::SetVAProcFilterDeinterlaceDefaultValue() {
+  deinterlace_caps_.mode_ = VAProcDeinterlacingNone;
```
Comment (kalyankondapally) [inline common/compositor/va/varenderer.cpp]: You want to check if it's not none already before setting update_Caps to true

File: common/compositor/va/varenderer.cpp [hunk 2]
```diff
@@ -143,6 +149,49 @@ bool VARenderer::SetVAProcFilterColorValue(HWCColorControl mode,
...
+  } else {
+    // TODO:Need video buffer meta data to judge if the frame really need
+    // Deinterlace.
+  }
+  return false;
+}
+bool VARenderer::SetVAProcFilterDeinterlaceValue(HWCDeinterlaceFlag flag,
+                                                 HWCDeinterlaceControl mode) {
+  if (flag == HWCDeinterlaceFlag::kDeinterlaceFlagNone) {
+    deinterlace_caps_.mode_ = VAProcDeinterlacingNone;
```
Comment (kalyankondapally) [inline common/compositor/va/varenderer.cpp]: You can call SetVAProcFilterDeinterlaceDefaultValue ?

**Pull request closed without merging**